### ISSUE TITLE
Missing Occupied Cells When Building Octree with `ocnn-pytorch`

### ISSUE THREAD

**Issue by kankannali** [octree-nn/ocnn-pytorch#45]
Hello O-CNN Development Team,

I hope this message finds you well. I am currently working with the `ocnn-pytorch` library and have encountered an issue related to octree construction and feature extraction. I would appreciate your assistance in resolving this matter.

**Environment**:
- **Operating System**: Ubuntu 20.04
- **Python Version**: Python 3.9
- **PyTorch Version**: PyTorch 2.2.2
- **ocnn-pytorch Version**: Latest from GitHub main branch (please specify the exact commit or tag if possible)
- **CUDA Version**: CUDA 12.4 (if applicable)

**Code Example**:
Below is a snippet of the code I am using to build the octree and extract the occupied space indices:

```python
import torch
import ocnn

# Construct point cloud data
points_tensor = torch.tensor([[1, 1, 1],
                             [2, 2, 2],
                             [10, 10, 10]], dtype=torch.float32)
features_tensor = torch.tensor([[0], [0], [0]], dtype=torch.float32)  # Example features

points = ocnn.octree.Points(points=points_tensor, features=features_tensor)
points = points.to('cuda')  # Assuming GPU usage

# Initialize octree with depth=5, full_depth=4, max_num=1
octree = ocnn.octree.Octree(depth=5, full_depth=4, max_num=1, device='cuda')
octree.build_octree(points)

# Retrieve occupied space cell indices
data = octree.get_input_feature("P", False)
print(data)

Issue: After running the above code, I expect the data output to contain three non-zero entries corresponding to the three input points: [1, 1, 1], [2, 2, 2], and [10, 10, 10]. However, the actual output only shows two non-zero entries, with the remaining entries being [0, 0, 0]. This suggests that one of the points is not being correctly represented in the octree.

**Comment by wang-ps** [comment]
The point cloud has to be normalized to [-1, 1]  before calling `octree.build_octree`. For more detail, please refer to the documents

<img width="567" alt="image" src="https://github.com/user-attachments/assets/88ef441a-1c2b-46fa-8afa-c3a9995b31ad" />

**Comment by kankanzheli** [comment]
I have a follow-up question regarding point cloud normalization and octree construction:
﻿
When constructing an octree within a fixed spatial boundary, I normalize the point cloud using the dimensions of this fixed space. If the objects within this fixed space are relatively small, should I include the extreme points of the fixed space as part of the point cloud to ensure that the octree properly partitions the space?
﻿
In other words, does incorporating boundary points help the octree maintain consistent spatial divisions, especially when dealing with small objects? Or I do not need to do it.
﻿
Your guidance on this matter would be greatly appreciated.
﻿
Thank you for your time and assistance.
